src/gui/folder.cpp

```diff
@@ -314,6 +314,62 @@ void Folder::etagRetreivedFromSyncEngine(const QString &etag)
     _lastEtag = etag;
 }
 
+void Folder::slotAccountCapabilitiesChanged()
+{
+    const Capabilities& capabilities = _accountState->account()->capabilities();
+    QString wsUrl = capabilities.getWebSocketUrl();
+    if (wsUrl.isEmpty()) {
+        return;
+    }
+
+    if (_webSocket) {
+        return;
+    }
+    //initialize wamp connection
+    _webSocket.reset(new QTcpSocket());
+    QUrl webSocketUrl(wsUrl);
+    QObject::connect(_webSocket.data(), &QTcpSocket::connected, [&]() {
+
+        QString sessionName("sessionName");
+        _session.reset(new QWamp::Session(sessionName, *_webSocket.data(), QWamp::Session::MessageFormat::Msgpack, true));
+
+        QObject::connect(_session.data(), &QWamp::Session::joined, [&](qint64 s) {
+            qDebug() << "Session joined to realm1 with session ID " << s;
```

Comment:
I know this is only temporary and you are going to clean up. In the final PR there should be no qDebug without categories.